### PR TITLE
✅ : make seleniums tests work again

### PR DESCRIPTION
**Pull request by juwit**
* makes selenium tests work again
* add a small overlay on the dashboard to mask widgets when data is loading (used by selenium to wait for the data to be rendered in the widgets instead of the default data '0')
* sort jobs by starting date

### CODE REVIEW

File: src/test/java/io/codeka/gaia/e2e/DashboardPage.kt [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
+        val wait = WebDriverWait(webDriver, 10)
+        wait.until(ExpectedConditions.titleIs("Gaia - Dashboard"))
+        wait.until(ExpectedConditions.numberOfElementsToBe(By.className("b-overlay"), 0))
+    }
+
+    @FindBy(xpath = "(//*[contains(@class,'widget-value')])[1]")
     private lateinit var modulesCountElement: WebElement
 
-    @FindBy(className="stacks_count")
+    @FindBy(xpath = "(//*[contains(@class,'widget-value')])[2]")
```
Comment (cdubuisson) [inline src/test/java/io/codeka/gaia/e2e/DashboardPage.kt:28]: We should use id selectors instead of xpath ones which seem to be more difficult to maintain

Comment (juwit) [reply src/test/java/io/codeka/gaia/e2e/DashboardPage.kt:28]: Oh yeah, I remember why I did that.
To use an id selector in this case, we have to pass the id as a prop to the component, as the value we want is located in a sub node. I thought it was too complex for a component to expose a property in its interface, just for the sake of selenium tests. So I choose to use the 'widget-value' class with a xpath selector.
There's also a bug with Kotlin in Selenium, which doesn't understand Kotlin's covariance.
In Java, Selenium can inject a `List<WebElement>`.
In Kotlin, when we declare a `List<WebElement>`, it is in fact a `List<? extends WebElement>`, and Selenium doesn't like it. I may open an issue for that.
Anyway, I'll use something else here :)

Comment (juwit) [reply src/test/java/io/codeka/gaia/e2e/DashboardPage.kt:28]: For future reference, the problen with the `@FindBy` annotation and type covariance is here :

org.openqa.selenium.support.pagefactory.DefaultFieldDecorator#isDecoratableList

Comment (juwit) [reply src/test/java/io/codeka/gaia/e2e/DashboardPage.kt]: ✅

File: src/test/java/io/codeka/gaia/e2e/SeleniumIT.java [hunk 2]
```diff
@@ -28,10 +28,9 @@
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @DirtiesContext
 @Testcontainers
-
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 @Tag("e2e")
-@Disabled("Disabled until complete migration of client")
+//@Disabled("Disabled until complete migration of client")
```
Comment (cdubuisson) [inline src/test/java/io/codeka/gaia/e2e/SeleniumIT.java:33]: Just remove the line and all is good :+1:

**Pull request merged**